### ISSUE TITLE
ctrl-. (ctrl-dot) not working as expected inside tmux

### ISSUE THREAD

**Issue by mohan43u** [tmux/tmux#4188]
### Issue description

when I run `cat` command, pressing `ctrl-.` (ctrl - dot) produce `^[[27;5;46~` output, but when I reproduce this step inside tmux, I get only `.`

If I do `set -s extended-keys always` in tmux.conf, I'm getting `^[[27;5;46~` as expected for `ctrl-.`, but `ctrl-alt-n` produce `^[[27;7;110~` inside tmux instead of `^[^N`

### Required information

* `tmux -V` : tmux 3.5a
* `uname -sp`: Linux unknown
* `echo $TERM`: foot
* `tmux kill-server; tmux -vv new`
[tmux-server-5919.log](https://github.com/user-attachments/files/17382989/tmux-server-5919.log)
[tmux-out-5919.log](https://github.com/user-attachments/files/17382991/tmux-out-5919.log)
[tmux-client-5917.log](https://github.com/user-attachments/files/17382992/tmux-client-5917.log)

**Comment by nicm** [comment]
I do not see the `extended-keys` option being changed in this log.

**Comment by nicm** [comment]
OK the problem is that in `input_key_mode1` we do not handle any cases for both Ctrl and Meta, only Ctrl OR Meta, so `C-M-.` is left as the mode2 form.

I'm not really sure which keys we should handle here, @crzcrz do you know?

**Comment by crzcrz** [comment]
The problem here, as I see it, is that every terminal interprets the (whatever we have for) standards in its own way. tmux, however, does not pass the keys through transparently, but does remapping into the output modes according to its own reading of standards.

I made the implementation on the tables listed at xterm homepage, the way I understood them. This understanding could have been wrong of course.

I'm not sure what to do here, if we fix it one group of users, we are likely to break it for another.

![image](https://github.com/user-attachments/assets/a86221f7-0609-4a6e-9c68-9e40f8bccc94)

**Comment by crzcrz** [comment]
`C-.` has no representation in VT10x mode, so mode1 or 2 is required to handle it. As for `C-M-n`, any application supporting extended keys should have no problems handling either `^]^N` or `^[[27;7;110~`. Similarly, if it's a bash or zsh binding (for example), the user can bind both variants.

**Comment by nicm** [comment]
Actually, I'm wrong... iTerm2 does send `^[[27;7;46~` for `C-M-n` in mode 1, so that is fine.

The OPs original other example was `C-M-n` where he is expecting `^[^N`, I'm not so sure about that one.

**Comment by nicm** [comment]
I suspect that if `C-x` for any `x` where the key would generate a single character (`^N`, `n`, etc) should be mapped to `^[x` in mode 1.

**Comment by crzcrz** [comment]
I agree. If there's an unambiguous representation in the standard mode, let's emit it in mode 1 as well. I'll make a patch in the evening.

**Comment by nicm** [comment]
Brilliant, thanks!

**Comment by nicm** [comment]
This looks good to me but it is difficult to test because both the terminals I have here (iTerm2 and xterm) are sending `\033[27;5;110~` for both `C-n` and  `C-M-n` in mode 2 rather than `\033[27;7;110~`... it's weird because they both get `M-n` correct, I wonder if it's a macOS thing.

**Comment by crzcrz** [comment]
You can force iTerm into `CSI u` mode when tmux is already running (from the right-click menu). `C-M-` keys will be reported correctly then and can be tested.

**Comment by nicm** [comment]
Good stuff, this seems to work for me - I've applied it (to OpenBSD) now, let me know if you see any other problems. Thanks!
